Jenkinsfile2

```diff
@@ -236,63 +248,96 @@ ansiColor('xterm') {
+                    dir(publishS3Dir) {
+                        checkout scm
+                        unstash 'CHANGELOG'
+                    }
+                    project.pullFilesFromArch('build/dist/**/*.whl, build-sdist/dist/**/*.tar.gz', "${publishS3Dir}/dist")
 
-                    def targetFolder = 'nightly'
-                    if (isRelease()) {
-                        targetFolder = 'stable'
+                    final def publishPublicS3StageName = 'Publish to public S3'
+                    stage(publishPublicS3StageName) {
+                        dumpInfo()
+                        docker.withRegistry("https://docker.h2o.ai", "docker.h2o.ai") {
+                            withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "awsArtifactsUploader"]]) {
+                                docker.image("docker.h2o.ai/s3cmd").inside {
+                                    sh """
+                                        s3cmd put -P ${publishS3Dir}/dist/*.whl s3://h2o-release/datatable/${S3_FOLDER_STABLE}/datatable-${versionText}/
+                                        s3cmd put -P ${publishS3Dir}/dist/*.tar.gz s3://h2o-release/datatable/${S3_FOLDER_STABLE}/datatable-${versionText}/
+                                    """
+                                }
+                            }
+                        }
                     }
 
-                    docker.withRegistry("https://docker.h2o.ai", "docker.h2o.ai") {
-                        withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', credentialsId: "awsArtifactsUploader"]]) {
-                            docker.image("docker.h2o.ai/s3cmd").inside {
-                                project.pullFilesFromArch('build/dist/**/*.whl, build-sdist/dist/**/*.tar.gz', "${publishS3Dir}/dist")
-                                unstash 'VERSION'
-                                sh "echo 'Stashed files:' && cd ${publishS3Dir} && find dist"
-                                def versionText = utilsLib.getCommandOutput("cat build/dist/VERSION.txt")
-                                sh """
-                                   ls
-                                   s3cmd put -P ${publishS3Dir}/dist/*.whl s3://h2o-release/datatable/${targetFolder}/datatable-${versionText}/
-                                   s3cmd put -P ${publishS3Dir}/dist/*.tar.gz s3://h2o-release/datatable/${targetFolder}/datatable-${versionText}/
-                               """
+                    stage('GitHub Release') {
+                        dumpInfo()
+                        docker.withRegistry("https://docker.h2o.ai", "docker.h2o.ai") {
+                            docker.image('docker.h2o.ai/opsh2oai/hub').inside("--init -v /home/jenkins/.ssh:/home/jenkins/.ssh:ro -v /home/jenkins/.gitconfig:/home/jenkins/.gitconfig:ro") {
+                                withCredentials([string(credentialsId: 'h2o-ops-personal-auth-token', variable: 'GITHUB_TOKEN')]) {
```

Comment:
probably `H2O_OPS_TOKEN_CREDS_ID` (and even that variable name seems too specific, how about renaming it into just `CREDS_ID`?)